pkg/lib/authenticationflow/declarative/intent_check_conflict_and_create_identity.go

```diff
@@ -22,6 +22,26 @@ type IntentCheckConflictAndCreateIdenity struct {
 }
 
 var _ authflow.Intent = &IntentCheckConflictAndCreateIdenity{}
+var _ authflow.Milestone = &IntentCheckConflictAndCreateIdenity{}
+var _ MilestoneFlowCreateIdentity = &IntentCheckConflictAndCreateIdenity{}
+
+func (*IntentCheckConflictAndCreateIdenity) Milestone() {}
+func (*IntentCheckConflictAndCreateIdenity) MilestoneFlowCreateIdentity(flows authflow.Flows) (MilestoneDoCreateIdentity, authflow.Flows, bool) {
+	// If there are no conflicts, then we will have MilestoneDoCreateIdentity directly.
+	m, mFlows, ok := authflow.FindMilestoneInCurrentFlow[MilestoneDoCreateIdentity](flows)
+	if ok {
+		return m, mFlows, true
+	}
+
+	// If there are conflicts, then we delegate to MilestoneFlowAccountLinking, which also implements MilestoneFlowCreateIdentity.
+	// We cannot find MilestoneFlowCreateIdentity because that would find this intent, resulting in an infinite recursion.
+	m1, m1Flows, ok := authflow.FindMilestoneInCurrentFlow[MilestoneFlowAccountLinking](flows)
+	if ok {
+		return m1.MilestoneFlowCreateIdentity(m1Flows)
+	}
+
+	return nil, flows, false
+}
```

Comment:
Should `MilestoneDoCreateIdentity` and `MilestoneFlowCreateIdentity` be one thing?

I am thinking that it should conceptually be:
```
- IntentSignup
  - IntentCreateIdentity (MilestoneFlowCreateIdentity)
    - NodeDoCreateIdentity (MilestoneFlowCreateIdentity)
```

1. From IntentSignup, we find `MilestoneFlowCreateIdentity`, which get `IntentCreateIdentity`.

2. In `IntentCreateIdentity`, it cannot determine if the identity is created or not, so it continue to find `MilestoneFlowCreateIdentity` in the sub-nodes. Which get `NodeDoCreateIdentity`.

3. In `NodeDoCreateIdentity`, it is the one who create the identity, so he knows the identity is created or not, so it implements `MilestoneFlowCreateIdentity` with a correct return value.

Reply:
I did not do that because to IntentSignup, what matters to it is whether the create identity flow is spawned, whether the identity is created or not does not matter to IntentSignup.

I also want to keep MilestoneDoCreateIdentity simple. The presence of it means created. Your proposal needs to call the milestone to check if it is created or not. This is quite error-prone. If we fail to call the checking, we incorrectly assume it is created.

Reply:
I see, ok.